### PR TITLE
Components

### PR DESCRIPTION
**Pull request by larisagonchar**
Storybook is installed and opens.
All new components from task description are implemented as described, including optional behavior.
All components from task description have stories in Storybook and are visible when Storybook is opened.
Tests are written for all components and they all pass. Edge cases are covered.

### CODE REVIEW

File: src/components/context-menu/context-menu.jsx [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import React from 'react';
+import './context-menu.css';
+import closeImg from '../../../assets/close.png';
+
+const ContextMenu = ({ onClick }) => {
+  return (
+    <>
+      <div className='context-menu'>
+        <img className='context-menu__img' src={closeImg} alt='close' onClick={onClick} />
+        <div className='context-menu__item'>Edit</div>
```
Comment (SavouryGin) [inline src/components/context-menu/context-menu.jsx]: Давай использовать более семантическую верстку: для кнопок `button` элемент, для попапа можно попробовать `dialog`.

File: src/components/genre-select/genre-select.jsx [hunk 2]
```diff
@@ -1,22 +1,25 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './genre-select.css';
 import classNames from 'classnames';
 
 const GenreSelect = ({ genres, selectedGenre, onSelect }) => {
+  const [genre, setSelectedGenre] = useState(selectedGenre);
```
Comment (SavouryGin) [inline src/components/genre-select/genre-select.jsx]: Лучше придерживаться конвенции по именованию state переменных
`[variable, setVariable]`
в твоем случае
`[genre, setGenre]`

File: src/components/movie-details/movie-details.jsx [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+import searchImg from '../../../assets/search-button.png';
+import defaultImg from '../../../assets/no-image.png';
+import { formatTime } from '../../formatters/time.formatter';
+
+const MovieDetails = ({ selectedMovie }) => {
+  const [imageSrc, setImageSrc] = useState(selectedMovie.poster_path);
+
+  return (
+    <section className='movie-details'>
+      <div className='movie-details__header'>
```
Comment (SavouryGin) [inline src/components/movie-details/movie-details.jsx]: Для хедера семантически корректно использовать тег [header](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/header)

File: src/components/movie-details/movie-details.jsx [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+          </div>
+
+          <span className='movie-details__genre'>{selectedMovie.genres.join(', ')}</span>
+
+          <div className='movie-details__container'>
+            <span className='movie-details__year'>{selectedMovie.release_date}</span>
+            <span className='movie-details__time'>{formatTime(selectedMovie.runtime)}</span>
+          </div>
+
+          <span className='movie-details__description'>{selectedMovie.overview}</span>
```
Comment (SavouryGin) [inline src/components/movie-details/movie-details.jsx:31]: Этот большой компонент я бы разбил на подкомпоненты. Главный принцип Реакта - атомарность, компнонент должен отвечать за одну вещь, как функция в single responsibility принципе. В данном случае можно легко вынести компоненты
MovieHeader
MoviePoster
MovieGenre
MovieDate
MovieDescription
и положить их все в под папку `movie-details/elements`. Не бойся дробить компоненты на как можно более мелкие части. Со временем они все равно имеют свойство разростаться, а чем мельче компоненты, тем чище код и больше переиспользуемость. Например, MovieDescription, MovieGenre и возможно другие элементы потом можно будет переиспользовать в форме.

File: src/components/movie-tile/movie-tile.jsx [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+    <div data-testid='movie-tile' className='movie-tile' onClick={handleMovieTileClick}>
+      <img src={imageSrc} alt='not-found-poster' onError={() => setImageSrc(defaultImg)}/>
+
+      <div className='movie-tile__context-menu' onClick={handleContextMenuClick}>
+        <div></div>
+        <div></div>
+        <div></div>
+      </div>
+
+      {isContextMenuOpen && <ContextMenu onClick={handleContextMenuClick}/>}
```
Comment (SavouryGin) [inline src/components/movie-tile/movie-tile.jsx]: Я бы подумал над добавлением отдельного компонента для модального окна и по идее ContextMenu должно рендериться внутри него. Модалка за счет [портала](https://react.dev/reference/react-dom/createPortal) рендерится за пределами root элемента, ей можно задавать независимые стили и имплементировать focus trap. Можешь посмотреть пример тут
https://github.com/SavouryGin/react-ssr-movie-library/blob/main/src/client/components/modal-window/index.tsx

File: src/components/movie-tile/movie-tile.jsx [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+
+      <div className='movie-tile__info-container'>
+        <span className='movie-tile__info-title'>{movie.title}</span>
+        <span className='movie-tile__info-year'>{new Date(movie.release_date).getFullYear()}</span>
+      </div>
```
Comment (SavouryGin) [inline src/components/movie-tile/movie-tile.jsx]: InfoContainer тоже выглядит как кандидат для вынесения в отдельный компонент. Каждый раз, когда тебе хочется в JSX добавить пустую строку, чтобы отделить одну часть от другой, преодолевай это и выноси в отдельный компонент :)

File: src/components/sort-control/sort-control.jsx [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+    onSelect(e.target.value);
+  };
+
+  const options = SORT_CONTROLS.map(control => {
+    return <option value={control.value} key={control.id} className='sort-control__option'>{control.value}</option>;
+  })
+
+  return (
+    <div className='sort-control'>
+      <label className='sort-control__label'>Sort By</label>
```
Comment (SavouryGin) [inline src/components/sort-control/sort-control.jsx]: Лейбл должен быть связан со своим инпутом через аттрибут `htmlFor`.

File: src/tests/movie-details.test.js [hunk 8]
```diff
@@ -0,0 +1,20 @@
+import { render } from '@testing-library/react';
+import MovieDetails from '../components/movie-details/movie-details';
+
+describe('MovieDetails component', () => {
+  const movie = {
```
Comment (SavouryGin) [inline src/tests/movie-details.test.js]: Такие тестовые данные лучше складировать в специальную папку `src/__mocks__`, потом удобно переиспользовать в разных тестах.

**Review by SavouryGin — CHANGES_REQUESTED** [review]
Отличная работа! Я бы предложил немного доработать по комментам, пока время есть. Кроме того, у меня в стори буке взаимодействие с кнопками приводит к ошибкам, как будто эти функции не замоканы. Посмотри, пожалуйста

GenreSelect
image

SearchForm
image

File: src/components/context-menu/context-menu.jsx [hunk 9]
```diff
@@ -0,0 +1,18 @@
...
+
+const ContextMenu = ({ onClick }) => {
+  return (
+    <>
+      <div className='context-menu'>
+        <img className='context-menu__img' src={closeImg} alt='close' onClick={onClick} />
+        <Button type={'button'} title={'Edit'} className={'context-menu__button'} />
+        <Button type={'button'} title={'Delete'} className={'context-menu__button'} />
+      </div>
+    </>
```
Comment (SavouryGin) [inline src/components/context-menu/context-menu.jsx:14]: Ненужный фрагмент-обёртка

File: src/components/context-menu/context-menu.jsx [hunk 10]
```diff
@@ -0,0 +1,18 @@
+import React from 'react';
+import './context-menu.css';
+import closeImg from '../../../assets/close.png';
```
Comment (SavouryGin) [inline src/components/context-menu/context-menu.jsx:3]: Я бы порекомендовал настроить абсолютные импорты, чтобы не было `../../../`
https://www.bezkoder.com/absolute-import-react/

File: src/components/title/title.js [hunk 11]
```diff
@@ -0,0 +1,10 @@
+import React from 'react';
+import './title.css';
+
+const Title = ({ title }) => {
+  return (
+    <span className='title'>{title}</span>
```
Comment (SavouryGin) [inline src/components/title/title.js:6]: Ну вот такие компоненты это немного перебор :) Всё таки должна быть какая-то дополнительная логика и верстка здесь, не просто span. Хотя на будущее может и пригодться, если ты собираешься этот Title расширять.

Comment (SavouryGin) [reply src/components/title/title.js:6]: И опять же, семантика. То, что называется title, должны быть скорее всего h2 или h3, ну или хотя быть strong, не просто безликий span.

**Review by SavouryGin — APPROVED** [review]
Great job! Keep it going :+1:
